### PR TITLE
added a new function called middleware which is called before callback

### PR DESCRIPTION
**Pull request by spaladu5**
Created a new function called middleware, which are executed before the callbacks.

### CODE REVIEW

File: test/unit/Router.php
Comment (jgarc186) [inline test/unit/Router.php:1]: Looks like you copied the Router.php file to the test folder. This is not needed here, please remove this file from this foler.

File: src/Garcia/Router.php [hunk 1]
```diff
@@ -43,18 +56,20 @@ public static function resource(string $path, string $className)
...
      *
      * @param string $path - URL path
      * @param callable $handler - Route handler
-     * @return void
+     * @return object
      */
     public static function get(string $path, callable $handler)
     {
         self::addRoute('GET', $path, $handler);
+         return new static;
```
Comment (jgarc186) [inline src/Garcia/Router.php]: This is good, please at the `return new static` statement to all the other request: `post`, `put`, etc. and update the return comments accordingly. Thank you!

**Comment by spaladu5** [comment]
added return new static to all the request methods

File: test/unit/Router.php
Comment (spaladu5) [reply test/unit/Router.php:1]: Fixed it

File: src/Garcia/Router.php (hunk 1 above)
Comment (spaladu5) [reply src/Garcia/Router.php]: fixed it

File: test/unit/RouterTest.php [hunk 2]
```diff
@@ -69,13 +69,4 @@ public function testJsonResponse()
         $this->assertIsArray(Router::getRoutes());
         $this->assertCount(5, Router::getRoutes());
     }
-
-     /** @test - Test if the route is added to the routes array */
-     public function addResource()
```
Comment (jgarc186) [inline test/unit/RouterTest.php:74]: we need this test here, the resouece test is good. Please don't remove it. put it back.

Comment (spaladu5) [reply test/unit/RouterTest.php:74]: fixed it

**Pull request merged**